src/pss.jl

```diff
@@ -28,30 +42,57 @@ Construct SSDSimulator instance based on simulation
 
 Currently SSDSimulator does not have any parameters
 """
-function SSDSimulator(sim_conf::LegendGeSimConfig)
-    coord = haskey(sim_conf.dict.simulation, :coordinates) ? sim_conf.dict.simulation.coordinates : "cylindrical"
+# function SSDSimulator(sim_conf::LegendGeSimConfig)
+function SSDSimulator(simulation_settings::PropDict)
+    if(!haskey(simulation_settings, :crystal_metadata_path))
+        simulation_settings[:crystal_metadata_path] = ""
+        # simulation_settings.crystal_metadata_path = crystal_metadata_path
+        # ToDo: move somewhere else - irrelevant if only geometry is constructed, or when simulation read from cache
+        @warn "No crystal metadata path given. Simulation with dummy constant impurity density."
+    elseif !ispath(simulation_settings.crystal_metadata_path)
+        @error "The path to crystal metadata you provided is not valid! ($(simulation_settings.crystal_metadata_path))"
+    else
+        @info "Impurity profile information based on $(simulation_settings.crystal_metadata_path)"
+    end
```

Comment:
Yeah I made a note to "move somewhere else", but couldn't figure out a good solution yet. Because the idea was also, it's the same input for SSD or siggen, so then if I move it somewhere where it's constructed, it gotta be still "common" for SSD/siggen, not to copy paste double the same thing in two different functions

Reply:
The logic was also, the validity check should come before everything else, so that we don't waste time doing all the stuff that comes before impurity and only THEN realizing ah, we can't proceed